model/channel.go

```diff
@@ -188,6 +201,18 @@ func (o *Channel) IsValid() *AppError {
 	return nil
 }
 
+func (o *GetChannelsOptions) IsValid() *AppError {
+	if reflect.TypeOf(o.UserIds).String() != "[]string" {
+		return NewAppError("GetChannelsOptions.IsValid", "model.channel.is_valid.user.ids.app_error", nil, "", http.StatusBadRequest)
+	}
+
+	if reflect.TypeOf(o.ChannelTypes).String() != "[]string" {
+		return NewAppError("GetChannelsOptions.IsValid", "model.channel.is_valid.channel.types.app_error", nil, "", http.StatusBadRequest)
+	}
+
```

Comment:
Should the content of `ChannelTypes` and `UserIds` also be valided here? E.g. if the elements in `ChannelTypes` have the correct values like `model.CHANNEL*OPEN`.